src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -294,7 +294,9 @@ void FallBackCompute(FCompute fn, const nnvm::NodeAttrs &attrs,
 
   std::vector<TBlob> out_blobs(outputs.size());
   for (size_t i = 0; i < out_blobs.size(); i++) {
-    if (req[i] == kWriteTo)
+    // ensure output does not use mkldnn mem.
+    // for inplace, we already converted & copied input above.
+    if ((req[i] == kWriteTo) || (req[i] == kWriteInplace))
       const_cast<NDArray &>(outputs[i]).InvalidateMKLDNNData();
```

Comment:
Actually, we need to handle kAddTo properly here as well. If it's kAddTo, we can convert the layout of the output array to the default format directly. since it's output array, we expect its memory will be written.